builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -222,13 +227,19 @@ func (r *ResourceProvisioner) decodeConfig(c *terraform.ResourceConfig) (*Provis
 	}
 
 	if p.ValidationKeyPath != "" {
-		keyPath, err := homedir.Expand(p.ValidationKeyPath)
+		vkeyPath, err := homedir.Expand(p.ValidationKeyPath)
 		if err != nil {
 			return nil, fmt.Errorf("Error expanding the validation key path: %v", err)
 		}
-		p.ValidationKeyPath = keyPath
-	}
-
+		p.ValidationKeyPath = vkeyPath
+	}
+        if p.SecretKeyPath != "" {
+                skeyPath, err := homedir.Expand(p.SecretKeyPath)
```

Comment:
yea i had first read it from my phone, when i got on and made the changes i
saw what you meant.
Will this make the next release?

On Thu, Jul 9, 2015 at 9:46 AM, Sander van Harmelen <
notifications@github.com> wrote:

> In builtin/provisioners/chef/resource_provisioner.go
> <https://github.com/hashicorp/terraform/pull/2657#discussion_r34256048>:
>
> >  		if err != nil {
> >  			return nil, fmt.Errorf("Error expanding the validation key path: %v", err)
> >  		}
> > -		p.ValidationKeyPath = keyPath
> > -	}
> > -
> > +		p.ValidationKeyPath = vkeyPath
> > +	}
> > +        if p.SecretKeyPath != "" {
> > +                skeyPath, err := homedir.Expand(p.SecretKeyPath)
>
> Not sure if I understand your comment. My point here was about the keyPath
> variable being within separated scopes, so that you did not have to think
> of separate names for the variable in their own scopes.
>
> It's just a nit and I noticed you updated it in your latest commits, so
> looks fine now [image: :smiley:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hashicorp/terraform/pull/2657/files#r34256048>.
>